### PR TITLE
Replace libnacl with pynacl

### LINKED ISSUE #130: Integration stopped working with HA 2022.7.0
I read about changes in Python.

Status as of now: "Failed to set up: Import error

### PR DESCRIPTION
**Pull request by jeroen84**
Since libsodium is no longer included in the OS build of Home Assistant the libnacl library does no longer work out of the box.

With this PR the libnacl library is replaced by pynacl. Since not all functions used by the free@home integration are available in pynacl I have used two functions from the pure_pynacl library.

I have tested this change on HA 2022.8.4 Docker image on a Raspberry Pi.

Since I developed on a development image of Home Assistant, the code style is changed to Black as well for the applicable python files.

Closes #130

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,3 +93,4 @@ Many thanks to Tho85 for building future proof, function based components
 Thanks to Foti for testing the cover device!
 Thanks to Lasse Magnussen for the climate device!
 Thanks to Nadir for testing the weather station
+Thanks to jfindlay for making the the pure_pynacl library available: https://github.com/jfindlay/pure_pynacl
```
Comment (Tho85) [inline README.md:96]:
```suggestion
Thanks to jfindlay for making the the pure_pynacl library available: https://github.com/jfindlay/pure_pynacl
Thanks to jeroen84 for the PyNaCl implementation
```

**Review by Tho85 — COMMENTED** [review]
Nice! :tada: :rocket:

Works for me in my local Docker dev container. I will deploy this to my productive system and test it for a little bit longer.

A bit unfortunate that `pure_pynacl` is not available via pip and we have to include its source code.

**Comment by jheling** [comment]
Thanks! The changes work form me on a raspberry pi.

**Comment by Tho85** [comment]
Works fine for me too (on 2022.7.5, btw). Merging :rocket:

**Pull request merged**